frontend/public/components/utils/webhooks.tsx

```diff
@@ -14,58 +20,114 @@ enum TriggerTypes {
+    ];
+  }
+  return [
+    classNames('col-sm-2'),
+    classNames('col-sm-7', 'co-break-all'),
+    classNames('col-sm-3', 'hidden-xs'),
+    classNames('hidden'),
+  ];
+};
+
+export const WebhookTriggers: React.FC<WebhookTriggersProps> = props => {
+  const { resource } = props;
   const { name, namespace } = resource.metadata;
   const { triggers } = resource.spec;
+  const canGetSecret = useAccessReview({
+    group: SecretModel.apiGroup,
+    resource: SecretModel.plural,
+    verb: 'get',
+    namespace,
+  });
+  const tableColumnClasses = getTableColumnClasses(canGetSecret);
   const webhooks = _.filter(triggers, trigger => webhookTriggers.has(trigger.type));
   if (_.isEmpty(webhooks)) {
     return null;
   }
 
-  const getWebhookURL = trigger => {
+  const getWebhookURL = (trigger: WebhookTrigger, secret?: string) => {
     const triggerProperty = getTriggerProperty(trigger);
-
-    // FIXME: Consider showing the secret in the table so that users can copy the real URL. Maybe a show/hide toggle like secrets?
-    // const secret = _.get(trigger, [triggerProperty, 'secret'], '<secret>');
-    // return `${kubeAPIServerURL}/apis/build.openshift.io/v1/namespaces/${namespace}/buildconfigs/${name}/webhooks/${secret}/${triggerProperty}`;
-
-    return `${kubeAPIServerURL}/apis/build.openshift.io/v1/namespaces/${namespace}/buildconfigs/${name}/webhooks/<secret>/${triggerProperty}`;
+    return `${kubeAPIServerURL}/apis/build.openshift.io/v1/namespaces/${namespace}/buildconfigs/${name}/webhooks/${secret ? secret : '<secret>'}/${triggerProperty}`;
   };
 
-  const getSecretReference = trigger => {
+  const getSecretReference = (trigger: WebhookTrigger) => {
     const triggerProperty = getTriggerProperty(trigger);
+    if (_.has(trigger, [triggerProperty, 'secret'])) {
+      return;
+    }
+
     const secretName = _.get(trigger, [triggerProperty, 'secretReference', 'name']);
     return secretName
       ? <ResourceLink kind="Secret" name={secretName} namespace={namespace} title={secretName} />
       : <span className="text-muted">No secret</span>;
   };
 
+  const copyWebhookToClipboard = (trigger: WebhookTrigger) => {
+    const triggerProperty = getTriggerProperty(trigger);
+
+    // In case of obsolete `secret` field on the trigger.
+    // https://github.com/openshift/api/blob/master/build/v1/types.go#L950
+    if (_.has(trigger, [triggerProperty, 'secret'])) {
+      const webhookSecret = _.get(trigger, [triggerProperty, 'secret']);
+      const url = getWebhookURL(trigger, webhookSecret);
+      navigator.clipboard.writeText(url);
+      return;
+    }
+
+    const secretName = _.get(trigger, [triggerProperty, 'secretReference', 'name']);
+    k8sGet(SecretModel, secretName, namespace).then((secret: K8sResourceKind) => {
+      if (!_.has(secret, 'data.WebHookSecretKey')) {
+        errorModal({error: `Secret referenced in the ${triggerProperty} webhook trigger does not contain 'WebHookSecretKey' key.`});
```

Comment:
I’d add a comment saying that the webhook won’t work due to the invalid secret reference